cachito/workers/pkg_managers/general_js.py

```diff
@@ -118,23 +152,40 @@ def download_dependencies(request_id, deps, proxy_repo_url, skip_deps=None):
 
             if counter % batch_size == 0:
                 deps_batches.append([])
-            deps_batches[-1].append(dep_identifier)
+            deps_batches[-1].append((dep_identifier, external_dep_version))
             downloaded_deps.add(dep_identifier)
             counter += 1
 
         for dep_batch in deps_batches:
-            log.debug(f"Downloading the following npm dependencies: {', '.join(dep_batch)}")
-            npm_pack_args = ["npm", "pack"] + dep_batch
+            # Create a list of dependencies to be downloaded. Excluding 'external_dep_version'
+            # from the list of tuples
+            dep_batch_download = [i[0] for i in dep_batch]
+            log.debug(
+                f"Downloading the following npm dependencies: {', '.join(dep_batch_download)}"
+            )
+            npm_pack_args = ["npm", "pack"] + dep_batch_download
             output = run_cmd(npm_pack_args, run_params, "Failed to download the npm dependencies")
 
             # Move dependencies to their respective folders
             # Iterate through the tuples made of dependency tarball and dep_identifier
             # e.g. ('angular-animations-8.2.0.tgz', '@angular/animations@8.2.0')
             for dep_pair in list(zip(output.split("\n"), dep_batch)):
                 tarball = dep_pair[0]  # e.g. angular-animations-8.2.0.tgz
-                dir_path = dep_pair[1].rsplit("@", 1)[0]  # e.g. @angular/animations
+                dir_path = dep_pair[1][0].rsplit("@", 1)[0]  # e.g. @angular/animations
+                external_dep_version = dep_pair[1][1]
+
+                # In case of external dependencies, create additional intermediate
+                # parent e.g. github/<org>/<repo> or external-<repo>
+                extra_dir = ""
+                if external_dep_version is not None:
+                    match = re.match(r"(?P<prefix>[^:]+):?(?P<suffix>[^/]+)", external_dep_version)
```

Comment:
Why is the ':' between the prefix and suffix optional?

Also, is `git://$STRING` a valid value here? if so, the suffix captured would be just the ':' character, since since the suffix should be "any non-empty string after the prefix (or a ':') until a '/' is found.

Finally, note that if no match is found, re.match will return `None`.

Reply:
I added the `:` between prefix and suffix because for I/P `github:ReactiveX/rxjs#78032157f5c1655436829017bbda787565b48c30`, I need O/P as `ReactiveX/rxjs`. Without the added `:`, it's giving me `:ReactiveX/rxjs`.

`git://$STRING` is a valid value but in that case, it falls under the `other_prefixes` category and so we don't care about what `suffix holds`.

if `external_dep_version` is not None, there **must** be a match. IMO, we should just let it fail and the failure would be captured in https://github.com/release-engineering/cachito/pull/192/files#diff-c372cb74d95ccecd07e2b5b76a539cf3R157 even before reaching here if it's not a valid dependency. 
@mprahl Please correct me if I'm wrong!

Reply:
This is outdated. Resolving